### DISCUSSION TITLE
[IDEA] Background Image to Frames, Scrollbar to Frames & Support for Nuitka

### DISCUSSION THREAD

**Discussion by FarhanRah** [TomSchimansky/CustomTkinter#1156] [Ideas]
It would be great if three of these features could be implemented in future:

1) Add the functionality to add background image to a frame (without using a label so that any widgets placed on top of that image don't show a colored box around them but instead show the main background image)
2) Add a scrollbar to frames
3) Add support for Nuitka and cx_Freeze, I have noticed that application packaged by py_installer almost always gets flagged as virus by windows, so it would be great if other compilers were to be supported.

**Comment by VollGerne** [comment]
To your point one, you can set widgets as "transparent", that should do what you expect.
To your point 2: just add a CTKCanvas to the frame, bind the scrollbars to the canvas and add the scrollbars to the frame and you have what you want.

**Comment by FarhanRah** [comment]
I have tried to set each widgets fg_color="transparent" but unfortunately there still seems to be a black coloured box. I have included my code and an image as reference.

Code:
```
import customtkinter
import os
import sys
from PIL import Image


def resource_path(relative_path):
    base_path = getattr(
        sys,
        '_MEIPASS',
        os.path.dirname(os.path.abspath(__file__)))
    return os.path.join(base_path, relative_path)


image_path = resource_path("images")

root = customtkinter.CTk()
root.config(background='#000001')
root.geometry('1200x800')

navbar_image = customtkinter.CTkImage(Image.open(os.path.join(image_path, "navbar.png")), size=(1200, 76))

# Create the frame for navigation bar
nav_bar = customtkinter.CTkFrame(master=root, width=1200, height=76, corner_radius=0, border_width=1)
nav_bar.grid_propagate(0)
nav_bar.place(x=0, y=30)

navbar_background = customtkinter.CTkLabel(nav_bar, text="", image=navbar_image, compound="left", fg_color="transparent")
navbar_background.place(x=0, y=0)

logo = customtkinter.CTkLabel(nav_bar, text="  Application", compound="left", fg_color="transparent")
logo.grid(row=0, column=0, padx=20, pady=10)

portfolio_button = customtkinter.CTkButton(nav_bar, corner_radius=0, height=40, border_spacing=10, text="Portfolio", fg_color="transparent", text_color=("#47A663", "#47A663"), hover=False, font=("Inter Medium", 16))
portfolio_button.grid(row=0, column=1, padx=120)

root.mainloop()
```

Image:
![image](https://user-images.githubusercontent.com/71914057/216280726-762dfd4a-96f1-4a47-b77f-20f989fda29b.png)
